ksqldb-engine/src/main/java/io/confluent/ksql/topic/TopicDeleteInjector.java

```diff
@@ -100,26 +98,18 @@ public TopicDeleteInjector(
 
     if (source != null) {
       checkTopicRefs(source);
-      try {
-        ExecutorUtil.executeWithRetries(
-            () -> topicClient.deleteTopics(ImmutableList.of(source.getKafkaTopicName())),
-            ExecutorUtil.RetryBehaviour.ALWAYS);
-      } catch (final Exception e) {
-        throw new RuntimeException("Could not delete the corresponding kafka topic: "
-                + source.getKafkaTopicName(), e);
-      }
 
+      deleteTopic(source);
+
+      final Closer closer = Closer.create();
+      deleteValueSubject(source);
+      deleteKeySubject(source);
       try {
-        final Format valueFormat = FormatFactory
-            .fromName(source.getKsqlTopic().getValueFormat().getFormat());
-
-        if (valueFormat.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
-          SchemaRegistryUtil.deleteSubjectWithRetries(
-                  schemaRegistryClient,
-                  source.getKafkaTopicName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
-        }
+        closer.close();
```

Comment:
Is this functional?

Reply:
what do you mean by "functional"? If by "does it do anything" the answer is yes - it will first close all of the closeables registered, and then it'll throw the first non-suppressed error that it encountered

Reply:
Where are closeables registered?

Reply:
🤦 I think something got lost in my merge. Thanks for catching this...

Reply:
https://github.com/confluentinc/ksql/pull/6461